src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml

```diff
@@ -298,6 +298,31 @@ paths:
       requestBody:
         $ref: '#/components/requestBodies/UpdateInduction'
 
+  #
+  # Subject access request
+  # --------------------------------------------------------------------------------
+  '/subject-access-request':
+    get:
+      summary: Retrieves PLP SAR data
+      tags:
+        - Subject Access Request
+
+      parameters:
+        - $ref: "#/components/parameters/prisonReferenceNumber"
+        - $ref: "#/components/parameters/fromDate"
+        - $ref: "#/components/parameters/toDate"
+      responses:
+        '200':
+          $ref: '#/components/responses/SarContentFound'
+        '204':
```

Comment:
I know these are status codes from the controller in the spring boot starter, but they feel weird ...
I wonder what the difference between 204 (no content found) and 209 (subject not recognised) is, and crucially what we have to implement (what exceptions do we throw??) to result in those status codes?
It feels like we need these 2 cases as specific tests in the integration test class?

FWIW, to me both cases feel like a 404!
If the answer is "return a 204 when its a prisoner but we hold no data for them" and "return a 209 when its not a recognised prisoner"  I don't know how we will determine and differentiate those 2 cases 🤔

Reply:
Looking at the controller implementation, it looks like it will be a "noContent" response (204) if the service implementation returns null. The service implementation as written does this, so I think we could (and should) write an integration test for that 👍 

Re: it returning a 209, I can't see how that could ever be the case! I wonder if we should remove the 209 from our swagger spec because I honestly dont think we could ever return that kind of response!

```
  fun getSarContentByReference(
    @RequestParam(name = "prn")
    @Parameter(description = "NOMIS Prison Reference Number")
    prn: String?,
    @RequestParam(name = "crn")
    @Parameter(description = "nDelius Case Reference Number")
    crn: String?,
    @RequestParam(value = "fromDate")
    @Parameter(description = "Optional parameter denoting minimum date of event occurrence which should be returned in the response")
    fromDate: LocalDate?,
    @RequestParam(value = "toDate")
    @Parameter(description = "Optional parameter denoting maximum date of event occurrence which should be returned in the response")
    toDate: LocalDate?,
  ): ResponseEntity<Any> {
    if (prn.isNullOrBlank() && crn.isNullOrBlank()) {
      return ResponseEntity.badRequest().body(
        ErrorResponse(
          status = HttpStatus.BAD_REQUEST,
          userMessage = "One of prn or crn must be supplied.",
          developerMessage = "One of prn or crn must be supplied.",
        ),
      )
    }

    val content = if (service is HmppsPrisonSubjectAccessRequestService && !prn.isNullOrBlank()) {
      service.getPrisonContentFor(prn, fromDate, toDate)
    } else if (service is HmppsProbationSubjectAccessRequestService && !crn.isNullOrBlank()) {
      service.getProbationContentFor(crn, fromDate, toDate)
    } else if (service is HmppsPrisonProbationSubjectAccessRequestService) {
      service.getContentFor(prn, crn, fromDate, toDate)
    } else {
      return ResponseEntity.status(209).build()
    }

    return content?.let { ResponseEntity.ok(it) } ?: ResponseEntity.noContent().build()
  }
```

Reply:
I take that back - I **can** see how a 209 would be returned (but it took me many many reads of the controller code in the library! 🤣 )

Our service impl is an implementation of `HmppsPrisonSubjectAccessRequestService` and requires a `prn`
But the check re: returning a 400 bad request only returns 400 if **both** the `prn` and `crn` query string params are blank.

Therefore, it would be possible to call the API `/subject-access-request?crn=xxxxx` - that would pass the 400 bad request check, but then return a 209
I guess its kinda correct, in that our implementation requires a `prn` but you've not specified one on the query string. So saying the `Subject Identifier is not recognised by this service` is correct 🤷 

Interestingly a HTTP 209 appears to be a non-standard response code! 🤷‍♂️ 
https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml